File: examples/go/stack_1.go
package main

import "fmt"

type Stack []int

func (s *Stack) Push(v int) {
	*s = append(*s, v)
}

func (s *Stack) Pop() int {
	n := len(*s) - 1
	v := (*s)[n]
	*s = (*s)[:n]
	return v
}

func main() {
	var s Stack
	s.Push(1)
	s.Push(2)
	s.Push(3)
	fmt.Println(s.Pop(), s.Pop(), s.Pop())
}

File: examples/rust/stack_1.rs
struct Stack<T> {
    data: Vec<T>,
}

impl<T> Stack<T> {
    fn new() -> Self {
        Stack { data: Vec::new() }
    }
    fn push(&mut self, x: T) {
        self.data.push(x);
    }
    fn pop(&mut self) -> Option<T> {
        self.data.pop()
    }
}

fn main() {
    let mut s = Stack::new();
    s.push(1);
    s.push(2);
    println!("{:?}", s.pop());
    println!("{:?}", s.pop());
}
